### DISCUSSION TITLE
Timeout opening a remote file in docker app

### DISCUSSION THREAD

**Discussion by ClaudioBarcaTEORESI** [Kitware/trame#628] [Q&A]
Hello, 
my goal is to pass a remote file .vtu to a viewer onto a docker container.

The `viewervtu` run fine outside the docker, it can open the file and rendering it.
Using in docker I have changed to host url with the IP where the .vtu is published but the `request.get(url)` go to Timeout `Session did not start before timeout expired. Check session logs`.
``

A snippets of my code : 
```

# URL dell'API
# url = "http://localhost:5102/static/file2.vtu"             <<-- this work on standalone mode
url = "http://192.168.1.202:5102/static/file2.vtu"         <<-- this NOT work on docker container

# get the file from the URL
response = requests.get(url)

if response.status_code == 200:
    with tempfile.NamedTemporaryFile(delete=False, suffix=".vtu") as tmp_file:
        tmp_file.write(response.content)
        tmp_file_path = tmp_file.name
        reader = vtkIOXML.vtkXMLUnstructuredGridReader()
        reader.SetFileName(tmp_file_path)
        reader.Update()
        dataset = reader.GetOutput()
        ....
```

The `Dockerfile` is :
```
FROM kitware/trame
COPY --chown=trame-user:trame-user . /deploy
ENV TRAME_CLIENT_TYPE=vue2
RUN /opt/trame/entrypoint.sh build
```

and the `compose.yaml` is :
```
services:
    trame:
        container_name: pytrame
        image: pytrame
        ports:
            - 8082:80
```

Any help please ? Thank you in advance

**Comment by jourdain** [comment]
I have my idea of what could be going on, but to be sure, you should look at a session log in `/deploy/server/logs/launcher/*.txt`. I bet you have a VTK crash when it tries to do the rendering.

**Comment by clabnet** [comment]
@jourdain Thank you for your fast reply.
You are right. Is there an error on startup.
```
Traceback (most recent call last):
  File "/deploy/viewervtu.py", line 41, in <module>
    import requests
ModuleNotFoundError: No module named 'requests'
```
But I have tried many times to add the missing packages, and I still haven't succeeded, there is always something missing.
The setup/requirements.txt file at this moment is :
```
trame
trame-components
trame-server
trame-vuetify
trame-vtk
vtk
wslink
numpy
```
The question is this: which is the most suitable docker image to render .vtu files?

[Here](https://github.com/clabnet/pytrame-simplest) my test repo.

**Reply by jourdain** [reply]
So why is your root `requirements.txt` different than the one in `./setup/requirements.txt`? So far it seems that you are at least missing `requests`. 

Also rendering wise it depend where your docker is going to run. Do you have an nvidia gpu, does your docker have access to the gpu, do you need server side rendering? (as you are using `VtkLocalView`, it seems that you could be alright if you don't call Render())

As you can see you are getting into deployment details with possible hardware dependencies. Without context or information, I can not tell you which path is a viable one for your setup.

**Reply by clabnet** [reply]
I have simplified the test repo and now it contain a running docker configuration. Thank you @jourdain
